File: PYTHON PROGRAMS/SearchBenchmarks.py
import random

def main():
    integers = [random.randint(0, 100) for i in range(20)]
    print("Benchmark Array:", *integers)

    targetIndex = random.randint(0, len(integers))
    targetValue = integers[targetIndex]
    
    sequentialComparisons = sequentialSearch(integers, targetValue)
    binaryComparisons = binarySearch(integers, targetValue)
    
    print("Target Value:", targetValue)
    print("Sequential Search Algorithm Comparisons:", sequentialComparisons)
    print("Binary Search Algorithm Comparisons:", binaryComparisons)

def sequentialSearch(integers, targetValue):
    comparisons = -1
    for i in integers:
        comparisons += 1
        if i == targetValue:
            return comparisons
    return comparisons

def binarySearch(integers, targetValue):
    comparisons = -1
    first = 0 
    last = len(integers) - 1
    found = False
    while not found and first <= last:
        middle = (first + last) // 2
        comparisons += 1
        if integers[middle] == targetValue:
            found = True
            comparisons = middle
        elif integers[middle] > targetValue:
            last = middle - 1
        else: first = middle + 1
    return comparisons

main()

File: JAVA PROGRAMS/SearchBenchmarks.java
import java.util.Random;

public class SearchBenchmarks {
    public static void main(String[] args) {
        Random randomNumbers = new Random();
        
        int[] integers = new int[20];
        for (int i = 0; i < 20; i++) integers[i] = randomNumbers.nextInt(100);
        System.out.print("Benchmark Array: ");
        for (int i = 0; i < 20; i++) System.out.print(integers[i] + " ");
        
        int targetIndex = randomNumbers.nextInt(integers.length);
        int targetValue = integers[targetIndex];
        
        int sequentialComparisons = sequentialSearch(integers, targetValue);
        int binaryComparisons = binarySearch(integers, targetValue);
        System.out.println("");
        System.out.println("Target Value: " + targetValue);
        System.out.println("Sequential Search Algorithm Comparisons: " + sequentialComparisons);
        System.out.println("Binary Search Algorithm Comparisons: " + binaryComparisons);
    }
    public static int sequentialSearch(int[] integers, int targetValue) {
        int comparisons = -1;
        for (int i = 0; i < integers.length; i++) {
            comparisons++;
            if (integers[i] == targetValue) return comparisons;
        }
        return comparisons;
    }
    public static int binarySearch(int[] integers, int targetValue) {
        int comparisons = -1;
        int first = 0;
        int last = integers.length - 1;
        boolean found = false;
        
        while (!found && first <= last) {
            int middle = (first + last) / 2;
            comparisons++;
            if (integers[middle] == targetValue) {
                found = true;
                comparisons = middle;
            } else if (integers[middle] < targetValue) {
                first = middle + 1;
            } else {
                last = middle - 1;
            }
        }
        return comparisons;
    }
}